internal/util-collection/src/main/scala/sbt/internal/util/Attributes.scala

```diff
@@ -92,10 +92,8 @@ object AttributeKey {
       rank0: Int
   )(implicit mf: Manifest[T], ojw: OptJsonWriter[T]): AttributeKey[T] =
     new SharedAttributeKey[T] {
-      require(name.headOption match {
-        case Some(c) => c.isLower
-        case None    => false
-      }, s"A named attribute key must start with a lowercase letter: $name")
+      require(name.headOption.fold(false)(_.isLower),
```

Comment:
Pattern match is the most readable and most performant ("probably at least 10 times as fast as the alternatives" according to Martin).